go/genesis/grpc.go

```diff
@@ -35,72 +35,19 @@ type grpcServer struct {
 // ToGenesis generates a genesis document based on current state at given height.
 func (s *grpcServer) ToGenesis(ctx context.Context, req *pb.GenesisRequest) (*pb.GenesisResponse, error) {
 	height := req.GetHeight()
-	if height <= 0 {
-		var err error
-		if height, err = s.tendermintService.GetHeight(); err != nil {
-			return nil, err
-		}
-	}
-
-	blk, err := s.tendermintService.GetBlock(&height)
-	if err != nil {
-		return nil, err
-	}
 
-	// Get genesis doc.
-	genesisProvider, err := New()
-	if err != nil {
-		return nil, err
-	}
-	genesisDoc, err := genesisProvider.GetGenesisDocument()
-	if err != nil {
-		return nil, err
-	}
-
-	// Call ToGenesis on all backends and merge the results together.
-	epochtimeGenesis, err := s.epochtimeBackend.ToGenesis(ctx, height)
-	if err != nil {
-		return nil, err
-	}
-	registryGenesis, err := s.registryBackend.ToGenesis(ctx, height)
-	if err != nil {
-		return nil, err
-	}
-	roothashGenesis, err := s.roothashBackend.ToGenesis(ctx, height)
+	// Get tendermint state as a genesis doc.
+	genesisDoc, err := s.tendermintService.ToGenesis(ctx, height, s.keymanagerBackend, s.registryBackend, s.roothashBackend, s.stakingBackend, s.schedulerBackend)
```

Comment:
Once #1879 is done we should also change the grpc server to take `consensus.Backend` instead of `service.TendermintService` specifically? Or maybe we can do this already?

Reply:
good idea, can already do it yes